packages/config-loader/src/lib/reader.ts

```diff
@@ -67,9 +67,9 @@ export async function readConfigFile(
     const out: JsonObject = {};
 
     for (const [key, value] of Object.entries(obj)) {
-      const result = await transform(value, `${path}.${key}`);
-      if (result !== undefined) {
-        out[key] = result;
+      // undefined covers optional fields, and nulls are explicitly skipped
+      if (value !== undefined && value !== null) {
+        out[key] = await transform(value, `${path}.${key}`);
```

Comment:
We do need to still check if the result is undefined, if it is we don't want to assign it to the output

Reply:
ok, fixed!